File: java/android_level_7/app/src/main/java/com/bliepy/bucketlistapp/store/DatabaseUtil.java
package com.bliepy.bucketlistapp.store;

import android.arch.persistence.db.SupportSQLiteDatabase;
import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;
import android.support.annotation.NonNull;

@Database(entities = {BucketlistModel.class}, version = 2, exportSchema = false)
public abstract class DatabaseUtil extends RoomDatabase {


    private final static String NAME_DATABASE = "bucketlist.db";

    public abstract BucketlistDao bucketlistDao();

    private static volatile DatabaseUtil INSTANCE;

    public static DatabaseUtil getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (DatabaseUtil.class) {
                INSTANCE = Room.databaseBuilder(context,
                        DatabaseUtil.class, NAME_DATABASE)
//                         .addCallback(stubdata)
                        .fallbackToDestructiveMigration()
                        .build();
            }
        }
        return INSTANCE;
    }

    public static void closeConnection(){
        INSTANCE.close();
    }

    private static RoomDatabase.Callback stubdata =
            new RoomDatabase.Callback() {

                @Override
                public void onOpen(@NonNull SupportSQLiteDatabase db) {
                    super.onOpen(db);
                    new DatabasePopulation(INSTANCE).execute();
                }
            };
}

File: kotlin/level_5_task_2/app/src/main/java/com/bliepy/gamebacklog/common/DatabaseUtil.kt
package com.bliepy.gamebacklog.common

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.bliepy.gamebacklog.dao.GameDao
import com.bliepy.gamebacklog.model.GameEntity

@Database(entities = [GameEntity::class], version = 4, exportSchema = false)
abstract class DatabaseUtil : RoomDatabase() {

    abstract fun gameDao(): GameDao

    companion object {
        private const val DATABASE_NAME = "GAME_DATABASE_2"

        @Volatile
        private var INSTANCE: DatabaseUtil? = null

        fun getDatabase(context: Context): DatabaseUtil? {
            if (INSTANCE == null) {
                synchronized(this) {
                    if (INSTANCE == null) {
                        INSTANCE = Room.databaseBuilder(
                            context.applicationContext,
                            DatabaseUtil::class.java, DATABASE_NAME
                        ).allowMainThreadQueries().fallbackToDestructiveMigration()
                            .build()
                    }
                }
            }
            return INSTANCE
        }
    }
}